scripts/test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+set -x
+set -u
+
+SCRIPT_DIRECTORY="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+ROOT_DIRECTORY="${SCRIPT_DIRECTORY}/.."
+BUILD_DIRECTORY="${ROOT_DIRECTORY}/build"
+TEMPORARY_DIRECTORY="${ROOT_DIRECTORY}/temp"
+
+KEYCHAIN_PATH="${TEMPORARY_DIRECTORY}/temporary.keychain"
+ARCHIVE_PATH="${BUILD_DIRECTORY}/Fileaway.xcarchive"
+FASTLANE_ENV_PATH="${ROOT_DIRECTORY}/fastlane/.env"
+
+# TODO: Source the env file if it exists?
+
+if [ -f "$FASTLANE_ENV_PATH" ] ; then
+    echo "Sourcing .env..."
+    source "$FASTLANE_ENV_PATH"
+fi
+
+
+function build_scheme {
+    # Disable code signing for the build server.
+    # TODO: Enable test builds if possible using a locally generated signing key.
+    # export CODE_SIGN_IDENTITY=""
+    # export CODE_SIGNING_REQUIRED=NO
+    # export CODE_SIGNING_ALLOWED=NO
+    # export DEVELOPMENT_TEAM=""
+    xcodebuild \
+        -workspace Fileaway.xcworkspace \
+        -scheme "$1" \
+        clean \
+        build \
+        CODE_SIGN_IDENTITY="" \
+        CODE_SIGNING_REQUIRED=NO \
+        CODE_SIGNING_ALLOWED=NO | xcpretty
+}
+
+cd "$ROOT_DIRECTORY"
+
+# List the available schemes
+xcodebuild -workspace Fileaway.xcworkspace -list
+
+# Smoke test builds.
+
+build_scheme "FileawayCore iOS"
+build_scheme "FileawayCore macOS"
+build_scheme "Fileaway iOS"
+build_scheme "Fileaway macOS"
+
+# Build the macOS archive.
+
+# Clean up the build directory.
+if [ -d "$BUILD_DIRECTORY" ] ; then
+    rm -r "$BUILD_DIRECTORY"
+fi
+mkdir -p "$BUILD_DIRECTORY"
+
+# Create the a new keychain.
+if [ -d "$TEMPORARY_DIRECTORY" ] ; then
+    rm -rf "$TEMPORARY_DIRECTORY"
+fi
+mkdir -p "$TEMPORARY_DIRECTORY"
+security create-keychain -p "$TEMPORARY_KEYCHAIN_PASSWORD" "$KEYCHAIN_PATH"
+security set-keychain-settings -lut 21600 "$KEYCHAIN_PATH"
+
+# Determine the version and build number.
+# TODO: Get Xcode to synthesise these itself so that builds also work there.
+VERSION_NUMBER=`cat macos/version.txt | tr -d '[:space:]'`
+GIT_COMMIT=`git rev-parse --short HEAD`
+TIMESTAMP=`date +%s`
+BUILD_NUMBER="${GIT_COMMIT}.${TIMESTAMP}"
+
+# Import the certificates into our dedicated keychain.
+fastlane import_certificates
```

Comment:
Fastlane currently complains that it's being run without Bundler. It wants us to call:

```bash
 bundle exec fastlane import_certificates
```

Reply:
Decided to reduce complexity by dropping Bundler for the time being. Long-term I'd love to remove the dependency on Fastlane, so hopefully we'll get away with this.